From 8c646f8c9a630b278c4e943200da2d6e823e7b8e Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Fri, 6 Jun 2014 18:02:58 +0200
Subject: [PATCH 3/8] Add support for creating and mounting FAT filesystems

---
 imgcreate/fs.py | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/imgcreate/fs.py b/imgcreate/fs.py
index 3d88ce9..378b49b 100644
--- a/imgcreate/fs.py
+++ b/imgcreate/fs.py
@@ -535,6 +535,24 @@ class ExtDiskMount(DiskMount):
         self.__resize_filesystem(size)
         return minsize
 
+class FatDiskMount(DiskMount):
+    """A DiskMount object that is able to format/resize FAT filesystems."""
+    def __init__(self, disk, mountdir, fstype, blocksize, fslabel,
+                 rmmountdir=True, tmpdir="/tmp"):
+        DiskMount.__init__(self, disk, mountdir, fstype, rmmountdir)
+        self.blocksize = blocksize
+        self.fslabel = "_" + fslabel
+        self.tmpdir = tmpdir
+
+    def mount(self):
+        logging.info("Formating %s filesystem on %s" % (self.fstype, self.disk.device))
+        rc = subprocess.call(["/sbin/mkfs." + self.fstype, "-n", self.fslabel, self.disk.device])
+
+        if rc != 0:
+            raise MountError("Error creating %s filesystem" % (self.fstype,))
+
+        DiskMount.mount(self)
+
 class DeviceMapperSnapshot(object):
     def __init__(self, imgloop, cowloop):
         self.imgloop = imgloop
-- 
1.9.3

