--- livecd-tools-18.10/imgcreate/live.py	2012-09-06 16:38:20.000000000 -0700
+++ livecd-tools-18.10/imgcreate/live.py.new	2012-09-12 20:29:23.006686900 -0700
@@ -400,8 +400,7 @@
         return options
 
     def _get_required_packages(self):
-        return ["syslinux", "grub2-efi", "shim"] \
-               + LiveImageCreatorBase._get_required_packages(self)
+        return ["syslinux"] + LiveImageCreatorBase._get_required_packages(self)
 
     def _get_isolinux_stanzas(self, isodir):
         return ""
