--- livecd-tools-15.2/imgcreate/yuminst.py.orig	2011-01-04 16:42:09.000000000 -0600
+++ livecd-tools-15.2/imgcreate/yuminst.py	2011-01-04 16:42:23.000000000 -0600
@@ -138,7 +138,6 @@
             # takes a variable and substitutes like yum configs do
             option = option.replace("$basearch", rpmUtils.arch.getBaseArch())
             option = option.replace("$arch", rpmUtils.arch.getCanonArch())
-            option = option.replace("$releasever", yum.config._getsysver("/", "redhat-release"))
             return option
 
         repo = yum.yumRepo.YumRepository(name)
